server/chat-commands/moderation.ts

```diff
@@ -88,7 +86,7 @@ exports.commands = {
 			return this.errorReply(`User '${name}' is unregistered, and so can't be promoted.`);
 		}
 
-		let currentGroup = room.getAuth({id: userid, group: (Users.usergroups[userid] || ' ').charAt(0)});
+		const currentGroup = room.getAuth({id: user.id, group: (Users.usergroups[userid] || ' ').charAt(0)} as User);
```

Comment:
```suggestion
		const currentGroup = room.getAuth({id: userid, group: (Users.usergroups[userid] || ' ').charAt(0)});
```

Fix the call signature of `room.getAuth`. If this works, its signature should just be `{id: ID, group: GroupSymbol}` rather than `User`.